rpc/clients/typescript/test/ws_client_test.ts

```diff
@@ -195,6 +195,55 @@ blockchainTests.resets('WSClient', env => {
             });
         });
 
+        describe('#getOrdersForPageAsync', async () => {
+            it('properly makes paginated request and returns signedOrders', async () => {
+                const ordersLength = 10;
+                const orders = [];
+                for (let i = 0; i < ordersLength; i++) {
+                    orders[i] = await orderFactory.newSignedOrderAsync({});
+                }
+                const validationResults = await deployment.client.addOrdersAsync(orders);
+                expect(validationResults.accepted.length).to.be.eq(ordersLength);
+
+                // NOTE(jalextowle): The time returned by Date uses milliseconds, but
+                // the mesh timestamp only uses second. Multiplying the seconds timestamp
+                // by 1000 gives us a comparable value. We only try to ensure that this
+                // timestamp is approximately equal (within 1 second) because the server
+                // will receive the request slightly after it is sent.
+                const now = new Date(Date.now()).getTime();
+                let page = 0;
+                const perPage = 5;
+                // First request for page index 0
+                let response = await deployment.client.getOrdersForPageAsync(page, perPage);
+                assertRoughlyEquals(now, response.snapshotTimestamp * secondsToMs(1), secondsToMs(2));
+                expect(uuidValidate(response.snapshotID)).to.be.true();
+
+                let responseOrders = response.ordersInfos;
+
+                // Second request for page index 1
+                page = 1;
+                response = await deployment.client.getOrdersForPageAsync(page, perPage, response.snapshotID);
+                expect(uuidValidate(response.snapshotID)).to.be.true();
+
+                // Combine orders found in first and second paginated requests
+                responseOrders = [...responseOrders, ...response.ordersInfos];
+
+                // Verify that all of the orders that were added to the mesh node
+                // were returned in the two `getOrders` rpc response
+                for (const order of orders) {
```

Comment:
Maybe we should make this a function since this pattern is now used twice. Not a big deal tho

Reply:
I also think we should do this.